### ISSUE TITLE
Max length of STMTTRN calculated incorrectly 

### ISSUE THREAD

**Issue by chbndrhnns** [csingley/ofxtools#28]
In my understanding, the max length is valid after resolving the entities like &amp; which gives a valid length of 32 in this case.

ValueError: Can't set OFX.creditcardmsgsrsv1 to <Element u'CREDITCARDMSGSRSV1' at 0x1139ad890>: Can't set CCSTMTTRNRS.ccstmtrs to <Element u'CCSTMTRS' at 0x1139ada10>: Can't set CCSTMTRS.banktranlist to <Element u'BANKTRANLIST' at 0x1139adb10>: Can't set STMTTRN.name to Lidl Vertriebs GmbH &amp; Co.Dresden: 'Lidl Vertriebs GmbH &amp; Co.Dresden' is too long; max length=32

**Comment by csingley** [comment]
That is indeed odd.  Try this:
```
In [1]: s = 'Lidl Vertriebs GmbH & Co.Dresden'

In [2]: from ofxtools.Types import String

In [3]: val = String(32)

In [4]: val.convert(s)
Out[4]: 'Lidl Vertriebs GmbH & Co.Dresden'

In [5]: val = s
In [6]: val
Out[6]: 'Lidl Vertriebs GmbH & Co.Dresden'
```
Does that work for you?

Sorry you're having so much trouble.

**Comment by csingley** [comment]
You could also upload an extract from the OFX data file containing that part of the STMTTRN and let me look at it, or else edit ofxtools.Types.String.convert() to add print(type(value)); print(len(value)) etc. etc. before it tests self.length and throws an error.

If I recall correctly you're running Python 2.  Remember, I inserted kind of an ugly hack in line 15 of ofxtools.Types.String... `str` is actually `unicode` under Python 2. However, your error message is displaying an actual str.  This is likely related to the strange behavior you're seeing.

I should really clean this str/unicode confusion up anyway.

**Comment by chbndrhnns** [comment]
Adding 

```
from HTMLParser import HTMLParser
        string = HTMLParser().unescape(string)
```

to _groomstring() also solves the issue.

Test file: 

[entities.ofx.txt](https://github.com/csingley/ofxtools/files/1161179/entities.ofx.txt)

**Comment by csingley** [comment]
There we go.  FYI, you can prevent Github from unescaping your text by enclosing it with backticks (`)

`Lidl Vertriebs GmbH &amp; Co.Dresden`

This is part of the OFX spec I missed implementing (it's in section 2.3).  It belongs in Types.String.convert().  I pushed a fix to master; it works on your test file.

**Comment by chbndrhnns** [comment]
You are so quick! Fantastic!

**Comment by csingley** [comment]
This one's easy once you know the problem.  Thanks for reporting sir!

**Comment by chbndrhnns** [comment]
I would like to reopen this one for the following example:

```
<BANKTRANLIST>
  9     ¦   ¦ <DTSTART>20160607
  8     ¦   ¦ <DTEND>20171026
  7     ¦   ¦ <STMTTRN>
  6     ¦   ¦   <TRNTYPE>DEBIT
  5     ¦   ¦   <DTPOSTED>20171025
  4     ¦   ¦   <TRNAMT>-2.00
  3     ¦   ¦   <FITID>1434
  2     ¦   ¦   <NAME>Bahn only - L&apos;tur Tour Baden-Bad
  1     ¦   ¦   <MEMO>Bahn only . L&apos;tur Tour Baden.BadenDatum 24.10.2017 Zeit 21.02Kaufumsatz VISA CardEUR 2,00
46      ¦   ¦ </STMTTRN>
  1     ¦   </BANKTRANLIST>
```

Seems like `&apos;` stays escaped here...

**Comment by csingley** [comment]
Welllll.... your bank appears to be sending OFXv1 (i.e. SGML not XML), and the OFXv1 spec doesn't define &apos; your bank's not following the spec (cf. OFX v1.6 section 2.3.2.1) so the fix to Types.String.convert() didn't catch it.

That being said, banks break with the spec all the time, so my policy is to be lenient when it's not harmful.  It's really not a big deal to add other XML escape characters to Types.String.convert(), so I'll do that.

**Comment by csingley** [comment]
Fixed in [0766212](http://github.com/csingley/ofxtools/commit/0766212646aa52424302114c064693108df4cfbe)

Give that a try

**Comment by chbndrhnns** [comment]
Works perfectly fine with me!
